python/tvm/relay/frontend/tensorflow.py

```diff
@@ -237,7 +238,7 @@ def _infer_out_shapes(inputs, params):
 
 def _infer_shape(node, params=None):
     """A method to get the output shape of an intermediate node in the relay graph."""
-    out_type = ir_pass.infer_type(node)
+    out_type = _transform.OptimizeOnExpr(node, _transform.InferType())
```

Comment:
I don’t think we want to introduce OptimizeOnExpr helper. Instead, we force everything into module to module and manually

Reply:
yeah, this helper function internally doesn't module->module opts and returns the entry_func of the updated module. It is mainly used by the python unit tests and cases where only infer type is needed.

I can probably remove them and add it in the corresponding python unit tests. How do you think?

Reply:
That sounds good, we do not want to encourage the use of this API